frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -162,11 +159,36 @@ export const XYChart = ({
   const areXTicksHidden = settings.x.tick_display === "hide";
   const xLabelOffset = areXTicksHidden ? -style.axes.ticks.fontSize : undefined;
 
-  const tickValues = fixTimeseriesTicksExceedXTickCount(
+  const xTickValues = fixTimeseriesTicksExceedXTickCount(
     settings.x.type,
     xScale.scale,
     xTickCount,
   );
+
+  const xTickFormat = (value: string | number | Date) => {
+    return formatXTick(value, settings.x.type, settings.x.format);
+  };
+
+  // Recalculate some of the values that depends on `xTickValues`
+  xTicksDimensions = getXTicksDimensions(
+    xTickValues.map(coerceNumber).map(xTickFormat),
+    series,
+    settings.x,
+    style.axes.ticks.fontSize,
+  );
+  margin = calculateMargin(
```

Comment:
Why it is not possible to calculate `xTicksDimensions` based on `xTickValues` initially? Seems like there is no dependency to anything that has not been calculated: `xTickValues.map(coerceNumber).map(xTickFormat)`

Reply:
`xTickValues` depends on `xScale` which depends on `calculatedInnerWidth` which depends on `innerWidth` which depends on `margin` which depends on `xTicksDimensions`